test/powershell/Modules/Microsoft.PowerShell.Management/PSDrive.Tests.ps1

```diff
@@ -1,45 +1,137 @@
-Describe "Validate basic PSDrive Cmdlets" -Tags "CI" {
+Describe "Basic Alias Provider Tests" -Tags "CI" {
+    Context "Validate basic PSDrive Cmdlets" {
+        BeforeAll {
+            #just use same location as TestDrive for simplicity
+            $psDriveRoot = "TestDrive:"
+            $psDriveName = "PsTestDriveName"
+        }
+
+        BeforeEach {
+            New-PSDrive -Name $psDriveName -PSProvider FileSystem -Root $psDriveRoot > $null
+        }
+
+        AfterEach {
+            Remove-PSDrive -Name $psDriveName -ErrorAction SilentlyContinue
+        }
+
+        It "Create a new PSDrive" {
+            $newDrive = New-PSDrive -Name "NewDifferentPSDrive" -PSProvider FileSystem -Root $psDriveRoot
+            try {
+                $newDrive.Name | Should Be "NewDifferentPSDrive"
+                $newDrive.Root | Should Be (Convert-Path $psDriveRoot)
+            }
+            finally {
+                Remove-PSDrive -Name "NewDifferentPSDrive" -ErrorAction SilentlyContinue
+            }
+        }
+
+        It "Read data from a PSDrive" {
+            $driveProp = Get-ItemProperty ${psDriveName}:
+            $driveProp.PSDrive.Name | Should Be $psDriveName
+        }
+
+        It "Remove the PSDrive" {
+            $existsBefore = Test-Path "${psDriveName}:\"
+            Remove-PSDrive -Name ${psDriveName} -ea SilentlyContinue
+            $existsAfter = Test-Path "${psDriveName}:\"
+            $existsBefore | Should Be $true
+            $existsAfter | Should Be $false
+        }
+
+        It "Verify 'Used' and 'Free' script properties" {
+            $drive = Get-PSDrive -Name $psDriveName
+            $drive.Used -gt 0 | Should Be $true
```

Comment:
I changed it to just detecting if the variables exist.
I am not sure how to determine if the used/free properties are correct or not.